app/bundles/PageBundle/EventListener/BuilderSubscriber.php

```diff
@@ -405,8 +418,27 @@ public function onPageDisplay(Events\PageDisplayEvent $event)
                 $savePrefs = $this->renderSavePrefs($params);
                 $content   = str_ireplace(self::saveprefsRegex, $savePrefs, $content);
             }
-        }
 
+            $dom = new DOMDocument('1.0', 'utf-8');
+            $dom->loadHTML(mb_convert_encoding($content, 'HTML-ENTITIES', 'UTF-8'), LIBXML_NOERROR);
+            $xpath      = new DOMXPath($dom);
+            // If use slots
+            $divContent = $xpath->query('//*[@data-prefs-center="1"]');
+            if (!$divContent->length) {
+                // If use tokens
+                $divContent = $xpath->query('//*[@data-prefs-center-first="1"]');
+            }
+            for ($i = 0; $i < $divContent->length; ++$i) {
+                $slot    = $divContent->item($i);
+                $newnode = $dom->createElement('startform');
+                $slot->parentNode->insertBefore($newnode, $slot);
+                break;
```

Comment:
I'm having trouble understand what's going on with the DOMDocument code. Why is this in foreach if it will be executed only once?

Reply:
Prefs center is little mess.
We can use prefs center with html or tokens with many blocks (categories, segments, frequency...)
All blocks I marked with attribute data-prefs-center/data-prefs-center-first.
Then I try find first block and before it set $startform.
Because I got $startform. as HTML code, I create fictive element <startform></startform> by DOMDocument and then replace by str_replace with my <form> html
I know maybe it's little tricky, but I didn't find better way.